system/goos/max-user-watches_linux.go

```diff
@@ -0,0 +1,28 @@
+package goos
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/ironstar-io/tokaido/utils"
+)
+
+// CheckMaxUserWatches checks if max_user_watches is sufficient for Tokaid to work properly
+func CheckMaxUserWatches() {
+	const maxUserWatchesRec = 524288
+
+	m, _ := strconv.Atoi(utils.StdoutCmd("cat", "/proc/sys/fs/inotify/max_user_watches"))
+
+	if m < maxUserWatchesRec {
+		fmt.Printf(`
+--- ❗ WARNING ❗ ---
+Your systems's max_user_watches value [%d] is less than the recommended [%d]
+This is known to cause issues problems with the Unison sync service on Linux.
+
+Please see this article for instructions on increasing your max_user_watches
+value: https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers
+---
+`, m, maxUserWatchesRec)
+	}
+
+}
```

Comment:
Yeah, this limitation hurts. Is there going to be a way to get around this do you think?

Reply:
Not really. We could attempt to fix it automatically, but it is another one of those arguably not appropriate things to do in linux. 

a] the user base is so small that the effort required to automate might not really pay off
b] Linux users are unlikely to want their core system settings directly modified

So I think we maintain our philosophy of letting linux users take care of themselves. As a Linux user, I'm personally happy with that approach.

Reply:
I'm sure our users will tell us otherwise :D